_posts/2021-10-01-gs_00004_basic_shell_script.md

```diff
@@ -0,0 +1,318 @@
+  overlay_image: /assets/images/getting_started.jpg
+---
+
+<div class='note-normal'>
+<top4>Prerequisites</top4>
+<ul>
+    <li><a href="/categories/getting_started/terminal_basics">Terminal Basics I</a></li>
+    <li><a href="/categories/getting_started/terminal_basics2">Terminal Basics II</a></li>
+</ul>
+</div>
+
+We are glad you made it back to our next post. Is your head ready to learn a
+very cool thing called **Shell Script**? Let's go...
+
+In our very first post, we stated that using a terminal has many advantages over
+a graphical interface. To convince you, we introduced very basic but powerful
+commands that you can use to control your computer with a terminal. If you have
+no idea what a terminal is or if you forgot some commands, just use one of the
+two links above to go back to the previous posts. Even things like the
+autocompletion and the "\*" operator are reason enough for us to use the
+terminal, this might not convince everyone. So, we want to go one step further.
+Are you ready?
+
+<div style="overflow: auto;" markdown="1">
+<figure class="img-right">
+  <img src="/assets/images/gs_00004_josephine.jpg" alt="">
+</figure>
+
+Let's do some time traveling into the year 1886. There was a woman called
+Josephine. She loved to celebrate parties. There was just one issue with that.
+She hated to clean the dishes afterward. So, she tried to build a machine to do
+it automatically for her. Finally, she was the first person in the world to
+construct a working dishwasher.
+
+Even though we do not expect you to invent something so meaningful as a
+dishwasher, the idea behind the concept should come to your mind very often.
+Whenever you have to do something regularly you should ask yourself if it is
+possible to automate it. This is more or less what a **Script** does. It calls a
+sequence of commands for you. We want to give you two examples that might be
+useful in the real world. Before that, we have to clarify the term **Shell**.
+
+</div>
+
+# Shell, Terminal, Console ...
+
+All these terms are very confusing for beginners as well as advanced engineers.
+Regularly, they are used interchangeably. Let's try to make them as clear as
+possible. In the very beginning, computers were huge and usually had no user
+interface. You needed some extra **physical device** that was connected to the
+computer, which was called **Console**. In contrast, the **Terminal** is not
+something physical. It is simply a text input and output environment. So far so
+good. You have a physical device (Console) with a keyboard or something similar
+and kind of a program where you can type some text in (Terminal). But there is
+one step more. Something needs to read the text you are typing and **interpret**
+its meaning. The software that is doing this task is called **Shell**. If you
+enter a command like "ls", the shell _understands_ that you want to get the
+content of the current folder. It executes the command and returns the output to
+the terminal.
```

Comment:
I think the transition to modern computers is missing here.

Something that popped into my mind while reading this section: This whole topic shell/console/terminal might be a bit confusing for a total newbie. You could address this more in depth in a separate "bonus" post. And in the current post, simply explain that a shell is the program that reads your commands and runs them. I think, for the topic of this post, that would be sufficient.

Reply:
I moved it as bonus chapter to the end